api/util/csp-request.ts

```diff
@@ -0,0 +1,85 @@
+import { CostRequest, CspRequestType, CspRequest } from "../../models/cost-jobs";
+import { CspResponse, ProvisionRequest } from "../../models/provisioning-jobs";
+import { getConfiguration } from "../provision/csp-configuration";
+import { getToken } from "../../idp/client";
+import { logger } from "../../utils/logging";
+import axios from "axios";
+
+/**
+ * Make a request to an actual CSP implementation of the ATAT API
+ *
+ * @param request The input request (e.g., provisioning, cost)
+ * @returns the response from the CSP
+ */
+export async function cspRequest(request: CspRequestType<CostRequest | ProvisionRequest>): Promise<CspResponse> {
+  if (!request.body) {
+    return { code: 400, content: { details: "No request body provided" } };
+  }
+  const { targetCsp, portfolioId } = request.body;
+
+  if (!targetCsp || !portfolioId) {
+    logger.error("Request to CSP failed", { input: { targetCsp, portfolioId } });
+    return {
+      code: 400,
+      content: {
+        details: "No Target CSP or the Portfolio Id is not provided.",
+        request: request.body,
+      },
+    };
+  }
+  const baseUrl = (await getConfiguration(targetCsp.name))?.uri;
+  let url: string;
+
+  switch (request.requestType) {
+    case CspRequest.PROVISION:
+      url = `${baseUrl}/portfolios`;
+      break;
+    case CspRequest.COST:
+      url = `${baseUrl}/portfolios/${portfolioId}/cost`;
+      break;
+  }
+
+  if (!baseUrl || !baseUrl.startsWith("https://")) {
+    logger.error("Invalid CSP configuration (Demo)", {
```

Comment:
What's the "(Demo)" for here?

Reply:
Used it to differentiate the error, but not needed anymore.  Removed.